File: Leetcode/hashmap/1_Two Sum/Solution.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = dict();
        for i in range ( len(nums)):
            if nums[i] in map:
                return [map[nums[i]] ,i]
            else:
                map[target-nums[i]] = i
        return None;

File: Leetcode/hashmap/1_Two Sum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int, int> map;
        for(int i = 0; i<nums.size(); ++i){
            if(map.count(target-nums[i])>0){
                res.push_back(map[target-nums[i]]);
                res.push_back(i);
                return res;
            }
            map[nums[i]] = i;
        }
        return res;
    }
};
